### PR TITLE
Remove duplicated classes from test session, fix some to be able to run in parallel and not restarting 3rd party for each tests

### PR DESCRIPTION
**Pull request by olamy**
- ee10 mongo session test in parallel
- cleanup pom
- move common class MongoTestHelper to a tools module
- fix dependencies
- jetty-ee9-test-sessions run in parallel and use common code to start mongodb
- use common class for memcached
- use common class for memcached
- make jdbc tests parallel
- make jdbc session tests work in parallel
- infinispan tests in parellel
- rework infinispan tests to avoid copies of same class
- fix duplicate classes infinispan test, do no restart image for each test
- infinispan reuse classes for ee9 as well
- hazelcast tests in parallel, remove duplicated classes
- init infinispan only at startup and upgrade Docker image to 11.0.14.Final
- reuse code for gcloud test and start container only once then reuse it for parallel tests
- better naming
- extract more common classes
- remove commented content

### CODE REVIEW

File: tests/jetty-test-session-common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  <name>Tests :: Session Test Tools</name>
+  <packaging>jar</packaging>
+
+  <properties>
+    <bundle-symbolic-name>${project.groupId}.session.test.tools</bundle-symbolic-name>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>jakarta.servlet</groupId>
```
Comment (janbartel) [inline tests/jetty-test-session-common/pom.xml]: Can't be common if it depends on servlet 6 api?

File: jetty-integrations/jetty-infinispan/jetty-infinispan-remote-query/pom.xml [hunk 2]
```diff
@@ -9,7 +9,6 @@
   <name>Integrations :: Infinispan :: Remote with Querying</name>
   <properties>
     <bundle-symbolic-name>${project.groupId}.infinispan.remote.query</bundle-symbolic-name>
-    <infinispan.docker.image.version>11.0.9.Final</infinispan.docker.image.version>
```
Comment (janbartel) [inline jetty-integrations/jetty-infinispan/jetty-infinispan-remote-query/pom.xml:12]: where's the version of infinispan image configured now, I can't find it?

Comment (olamy) [reply jetty-integrations/jetty-infinispan/jetty-infinispan-remote-query/pom.xml:12]: top pom (https://github.com/eclipse/jetty.project/pull/9782/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R200). I even keep a comment maybe we can remove the comment now? :)

File: tests/jetty-test-session-common/pom.xml (hunk 1 above)
Comment (olamy) [reply tests/jetty-test-session-common/pom.xml]: I was not sure about this. but it's marked as provided and optional.
It's only few classes from servlet spec.
the real artifact version is used in each ee9/ee10 module

Comment (olamy) [reply tests/jetty-test-session-common/pom.xml]: changed

**Comment by janbartel** [comment]
@olamy is this ready for review?

**Comment by olamy** [comment]
@janbartel yes I fixed your first comments

**Review by janbartel — APPROVED** [review]

**Pull request merged**